From 956b8a0421b724a6cd5e1be8198cea7b14c60cfa Mon Sep 17 00:00:00 2001
From: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
Date: Fri, 11 Sep 2020 16:27:20 +0530
Subject: [PATCH] Disable AVB Check

This avoids the message "There's an internal problem with your
device. Contact your manufacturer for details." on some AVB 1.0
devices.

Tracked-On: None
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 core/java/android/os/Build.java     | 2 +-
 core/jni/android_os_VintfObject.cpp | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/java/android/os/Build.java b/core/java/android/os/Build.java
index 75ae1f09755..d7666e355df 100755
--- a/core/java/android/os/Build.java
+++ b/core/java/android/os/Build.java
@@ -1062,7 +1062,7 @@ public class Build {
     public static boolean isBuildConsistent() {
         // Don't care on eng builds.  Incremental build may trigger false negative.
         if (IS_ENG) return true;
-        if (IS_USERDEBUG) return true;
+//        if (IS_USERDEBUG) return true;
 
         if (IS_TREBLE_ENABLED) {
             // If we can run this code, the device should already pass AVB.
diff --git a/core/jni/android_os_VintfObject.cpp b/core/jni/android_os_VintfObject.cpp
index ee11b6162db..1d411119dea 100644
--- a/core/jni/android_os_VintfObject.cpp
+++ b/core/jni/android_os_VintfObject.cpp
@@ -109,7 +109,7 @@ static jint android_os_VintfObject_verify(JNIEnv* env, jclass, jobjectArray pack
         }
     }
     std::string error;
-    int32_t status = VintfObject::CheckCompatibility(cPackageInfo, &error);
+    int32_t status = VintfObject::CheckCompatibility(cPackageInfo, &error, vintf::CheckFlags::DISABLE_AVB_CHECK);
     if (status)
         LOG(WARNING) << "VintfObject.verify() returns " << status << ": " << error;
     return status;
-- 
2.21.0

